src/Microsoft.SqlTools.ServiceLayer/QueryExecution/SpecialAction.cs

```diff
@@ -0,0 +1,125 @@
+// 
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution
+{
+    /// <summary>
+    /// Class that represents a Special Action which occured by user request during the query 
+    /// </summary>
+    public class SpecialAction {
+        
+        #region Private Class variables 
+        private bool none;
+        private bool expectActualYukonXmlShowPlan;
+        private bool expectEstimatedYukonXmlShowPlan;
+
+        #endregion
+
+        /// <summary>
+        /// The type of XML execution plan that is contained with in a result set  
+        /// </summary>
+        public SpecialAction()
+        {
+            None = true;
+        }
+
+        #region Public Functions
+        /// <summary>
+        /// No Special action performed 
+        /// </summary>
+        public bool None {
```

Comment:
This code particularly has me confused. So if you set None to true, you get both fields set to false, but none set to true? I'm ultimately not sure what you're trying to do with this, but I think there's a better way to do it.